src/transformStateWithClones.js

```diff
@@ -6,8 +6,42 @@
  *
  * @return {Object[]}
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+function transformStateWithClones(state, transforms) {
+  const object = Object.assign({}, state);
+  const arr = [];
+  function addProperties(copy, additionalProps) {
+    for (const key in additionalProps) {
+      copy[key] = additionalProps[key];
+    }
+    arr.push(Object.assign({}, copy));
+  }
+  function removeProperties(copy, additionalProps) {
+    additionalProps.forEach(function(key) {
+      delete copy[key];
+    });
+    arr.push(Object.assign({}, copy));
+  }
+  function clearProperties(copy) {
+    for (const key in object) {
+      delete copy[key];
+    }
+    arr.push(Object.assign({}, copy));
+  }
+  transforms.forEach(function(item) {
+    const { operation, properties } = item;
+    switch (operation) {
+      case 'addProperties':
+        addProperties(object, properties);
+        break;
+      case 'removeProperties':
+        removeProperties(object, properties);
+        break;
+      case 'clear':
+        clearProperties(object);
```

Comment:
The `clearProperties` function modifies the `object` directly. Ensure that a new copy of the state is created before modifying it to avoid unintended side effects.